07.bowling_object/game.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  def initialize(marks)
+    @marks = marks.split(',')
+  end
+
+  def score
+    frames = create_frames
+    frames.each_with_index.sum do |frame, idx|
+      if idx < 9
+        frame.score + calc_bonus(frames, frame, idx)
+      else
+        frame.score
+      end
```

Comment:
`calc_bonus`側に条件分岐を持っていってここはスッキリさせたいかも。（単なる好みの問題です）

```suggestion
      frame.score + calc_bonus(frames, frame, idx)
```

Reply:
以下のように`calc_bonus`側に条件分岐を持っていったところ、rubocopに怒られてしまったので、`strike?`と`spare?`メソッドを修正してみました。ご確認よろしくお願いします。
```ruby
def calc_bonus(frames, frame, idx)
    next_frame = frames[idx + 1]
    after_next_frame = frames[idx + 2]

    if idx < 9
      if frame.strike? && next_frame.strike?
        bonus_shot = idx < 8 ? after_next_frame[0] : next_frame[1]
        10 + bonus_shot.point
      elsif frame.strike?
        next_frame.first_two_points
      elsif frame.spare?
        next_frame[0].point
      else
        0
      end
    else
      0
    end
  end
```
<img width="593" alt="スクリーンショット 2022-04-19 23 15 35" src="https://user-images.githubusercontent.com/88083085/164027415-d733993f-a939-449f-b1d0-64904fc42159.png">

Reply:
改めて見直したところ、`strike?`メソッドに条件を追記したことで、`calc_bonus`内に不要な条件分岐ができたので、修正しました。ご確認よろしくお願いします。
https://github.com/shucream-p/ruby-practices/pull/10/commits/341a8e1a3c83a4e4b094b01303d76725c62edb65